cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResources.java

```diff
@@ -0,0 +1,119 @@
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.microsoft.azure.management.compute.implementation.DiskEncryptionSetInner;
+import com.sequenceiq.cloudbreak.cloud.EncryptionResources;
+import com.sequenceiq.cloudbreak.cloud.azure.client.AzureClient;
+import com.sequenceiq.cloudbreak.cloud.azure.client.AzureClientService;
+import com.sequenceiq.cloudbreak.cloud.exception.CloudConnectorException;
+import com.sequenceiq.cloudbreak.cloud.model.CloudResource;
+import com.sequenceiq.cloudbreak.cloud.model.Platform;
+import com.sequenceiq.cloudbreak.cloud.model.Variant;
+import com.sequenceiq.cloudbreak.cloud.model.encryption.CreatedDiskEncryptionSet;
+import com.sequenceiq.cloudbreak.cloud.model.encryption.DiskEncryptionSetCreationRequest;
+import com.sequenceiq.cloudbreak.cloud.notification.PersistenceNotifier;
+import com.sequenceiq.common.api.type.CommonStatus;
+
+@Service
+public class AzureEncryptionResources implements EncryptionResources {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureEncryptionResources.class);
+
+    private static final Pattern ENCRYPTION_KEY_URL_PATTERN = Pattern.compile("https://([^.]+)\\.vault.*");
+
+    @Inject
+    private AzureClientService azureClientService;
+
+    @Inject
+    private AzureUtils azureUtils;
+
+    @Inject
+    private PersistenceNotifier persistenceNotifier;
+
+    @Override
+    public Platform platform() {
+        return AzureConstants.PLATFORM;
+    }
+
+    @Override
+    public Variant variant() {
+        return AzureConstants.VARIANT;
+    }
+
+    @Override
+    public CreatedDiskEncryptionSet createDiskEncryptionSet(DiskEncryptionSetCreationRequest diskEncryptionSetCreationRequest) {
+        String vaultName;
+        String vaultResourceGroupName;
+        String desResourceGroupName;
+        AzureClient azureClient = azureClientService.getClient(diskEncryptionSetCreationRequest.getCloudCredential());
+
+        Matcher matcher = ENCRYPTION_KEY_URL_PATTERN.matcher(diskEncryptionSetCreationRequest.getEncryptionKeyUrl());
+        if (matcher.matches()) {
+            vaultName = matcher.group(1);
+        } else {
+            throw new IllegalArgumentException("vaultName cannot be fetched from encryptionKeyUrl. encryptionKeyUrl should be of format - " +
+                    "'https://<vaultName>.vault.azure.net/keys/<keyName>/<keyVersion>'");
+        }
+        if (diskEncryptionSetCreationRequest.isSingleResourceGroup()) {
+            desResourceGroupName = diskEncryptionSetCreationRequest.getResourceGroup();
+            vaultResourceGroupName = desResourceGroupName;
+        } else {
+            throw new IllegalArgumentException("Customer Managed Key Encryption for managed Azure disks is supported for single resource group only.");
```

Comment:
single resource group only -> what is this means? I think we should meantion something like "if the datahub/datalake is in the same resource group where the vault is"